pkg/expiringcache/cache.go

```diff
@@ -166,14 +167,14 @@ func (e *expiringCacheImpl) GetOrSet(key, value interface{}) interface{} {
 			e.removeNoLock(key)
 			e.addNoLock(key, currValue)
 		}
-		return currValue
+		return currValue.(V)
 	}
 
 	e.addNoLock(key, value)
 	return value
 }
 
-func (e *expiringCacheImpl) getValue(key interface{}) interface{} {
+func (e *expiringCacheImpl[K, V]) getValue(key interface{}) interface{} {
 	value := e.mq.get(key)
```

Comment:
Would it make sense to make the `mappedQueue` generic too and have these functions also use generics instead of `interface{}`?

Reply:
Make sense but in a separated PR, aleternatively I thought about replacing our code with https://github.com/hashicorp/golang-lru/blob/main/expirable/expirable_lru.go